### PR TITLE
Deprecate  'program' debug configuration property

### PR DESCRIPTION
**Pull request by SounD120**
This PR will deprecate `program` debug configuration property. Users that still use this property will see the following message in their debug console:
image

`${workspaceRoot}` was renamed to `${workspaceFolder}` due to this [recommendation](https://code.visualstudio.com/docs/editor/variables-reference#_why-isnt-workspaceroot-documented).

[AB#689](https://dev.azure.com/vscode-webdiag-extensions/5c2c3798-7e49-49f6-b45f-77d9c69636c6/_workitems/edit/689)

### CODE REVIEW

**Comment by ruslan-bikkinin** [comment]
@SounD120 please, also, update configurations in smoke tests.

**Comment by SounD120** [comment]
Sure, I want to make sure that both "program" and "cwd" parameters will work correctly.

File: package.nls.json [hunk 1]
```diff
@@ -18,6 +18,7 @@
     "reactNative.snippets.attachPackager.description":"A new configuration for attaching to packager",
     "reactNative.snippets.debugExpo.description":"A new configuration for launching Expo app",
     "reactNative.attach.program.description":"The path to launchReactNative.js in the vscode folder",
```
Comment (ruslan-bikkinin) [inline package.nls.json]:
```suggestion
    "reactNative.attach.program.description":"[DEPRECATED. USE cwd INSTEAD] The path to launchReactNative.js in the vscode folder",
```

File: package.nls.json [hunk 2]
```diff
@@ -29,6 +30,7 @@
     "reactNative.attach.debuggerWorkerUrlPath.description": "Path to the app debugger worker to override. For example, if debugger tries to attach to http://localhost:8081/debugger-ui/debuggerWorker.js and you get 404 error from packager output then you may want to change debuggerWorkerUrlPath to another value suitable for your packager (\"debugger-ui\" will be replaced with the value you provide).",
     "reactNative.launch.platform.description":"The platform to target",
     "reactNative.launch.program.description":"The path to launchReactNative.js in the vscode folder",
+    "reactNative.launch.cwd.description":"The path to project root folder",
```
Comment (ruslan-bikkinin) [inline package.nls.json]:
```suggestion
    "reactNative.launch.cwd.description":"[DEPRECATED. USE cwd INSTEAD] The path to project root folder",
```

File: src/debugger/nodeDebugWrapper.ts [hunk 3]
```diff
@@ -149,6 +146,12 @@ export function makeSession(
             } else {
                 logger.setup(Logger.LogLevel.Log, chromeDebugCoreLogs || false);
             }
+            if (args.program) {
+                // Remove this warning when program property will be completely removed
```
Comment (ruslan-bikkinin) [inline src/debugger/nodeDebugWrapper.ts]:
```suggestion
                // TODO: Remove this warning when program property will be completely removed
```

File: src/debugger/nodeDebugWrapper.ts [hunk 4]
```diff
@@ -150,6 +147,18 @@ export function makeSession(
...
             }
 
+            if (args.program) {
+                // TODO: Remove this warning when program property will be completely removed
+                logger.warn(localize("ProgramPropertyDeprecationWarning", "Launched debug configuration contains 'program' property which is deprecated and will be removed soon. Please replace it by: \"cwd\": \"${workspaceFolder}\""));
+                const useProgramEvent = TelemetryHelper.createTelemetryEvent("useProgramProperty");
+                Telemetry.send(useProgramEvent);
+            }
+            if (args.cwd) {
+                // To match count of 'cwd' users with 'program' users. Remove when program property will be removed
```
Comment (SounD120) [inline src/debugger/nodeDebugWrapper.ts]:
```suggestion
                // To match count of 'cwd' users with 'program' users. TODO: Remove when program property will be removed
```

File: src/debugger/nodeDebugWrapper.ts [hunk 5]
```diff
@@ -150,6 +147,18 @@ export function makeSession(
                 logger.setup(Logger.LogLevel.Log, chromeDebugCoreLogs || false);
             }
 
+            if (args.program) {
+                // TODO: Remove this warning when program property will be completely removed
+                logger.warn(localize("ProgramPropertyDeprecationWarning", "Launched debug configuration contains 'program' property which is deprecated and will be removed soon. Please replace it by: \"cwd\": \"${workspaceFolder}\""));
```
Comment (SounD120) [inline src/debugger/nodeDebugWrapper.ts]:
```suggestion
                logger.warn(localize("ProgramPropertyDeprecationWarning", "Launched debug configuration contains 'program' property which is deprecated and will be removed soon. Please replace it with: \"cwd\": \"${workspaceFolder}\""));
```

**Pull request merged**